src/data/data.cc

```diff
@@ -724,11 +724,15 @@ void MetaInfo::SynchronizeNumberOfColumns() {
 namespace {
 template <typename T>
 void CheckDevice(std::int32_t device, HostDeviceVector<T> const& v) {
-  CHECK(v.DeviceIdx() == Context::kCpuId || device == Context::kCpuId || v.DeviceIdx() == device)
-      << "Data is resided on a different device than `gpu_id`. "
-      << "Device that data is on: " << v.DeviceIdx() << ", "
-      << "`gpu_id` for XGBoost: " << device;
+  bool valid =
+      v.DeviceIdx() == Context::kCpuId || device == Context::kCpuId || v.DeviceIdx() == device;
+  if (!valid) {
+    LOG(FATAL) << "Invalid data ordinal, data is resided on a different device than the ordinal of "
+                  "the booster. The ordinal of the data is: "
+               << v.DeviceIdx() << ", the ordinal for XGBoost is: " << device;
```

Comment:
```suggestion
    LOG(FATAL) << "Invalid device ordinal, data is associated with a different device ordinal than "
                  "the Booster object. The device ordinal of the data is: "
               << v.DeviceIdx() << "; the device ordinal of the Booster object is: " << device;
```

Users may not understand what `ordinal` is. Let's use `device ordinal` to make it clear.